iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java

```diff
@@ -1456,6 +1456,63 @@ public void testCreateTableWithMetadataLocation() throws IOException {
             HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS.stream()).collect(Collectors.toList()), records, 0);
   }
 
+  @Test
+  public void testAlterTableWithMetadataLocation() throws IOException {
+    Assume.assumeTrue("Alter table with metadata location is only supported for Hive Catalog tables",
```

Comment:
What happens in the code where we try to do this for a non-HiveCatalog table?

Reply:
They will see no errors, the property will be changed on the HMS table, but the current metadata won't change.

Reply:
Seems like an inconsistent state.
Is there a way to recover from that, or shall we throw an exception instead?

Reply:
I'm not sure this is inconsistent. The HMS table property `metadata_location` has a special meaning if the table belongs to the hive catalog. It is also treated differently in our `HiveTableOperations` class, refreshing the metadata based on the value of this property.
In the case of tables from other catalog types, this is just a regular table property and it doesn't have any highlighted meaning. So I run an alter query I would expect that the table property is changed but nothing else.